model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/nh_solve/solve_nonhydro.py

```diff
@@ -1842,6 +2461,27 @@ def run_corrector_step(
                 offset_provider={},
             )
         log.debug(f"corrector start stencil 55")
+        """
+        rho (0:nlev-1):
+            Update the density at full levels (cell center) from the mass conservation equation (see eq. 3.21 in ICON tutorial 2023).
+            rho^{n+1} = rho^{n} - dt ( divergence(v^{n+1*} rho^{n}) + vwind_expl_wgt ( rho^{n}_{k-1/2} w^{n}_{k-1/2} - rho^{n}_{k+1/2} w^{n}_{k+1} ) / dz_{k} ) - dt * vwind_impl_wgt ( rho^{n}_{k-1/2} w^{n+1}_{k-1/2} - rho^{n}_{k+1/2} w^{n+1}_{k+1} ) / dz_{k} )
+            rho^{n+1} = z_rho_expl - dt * vwind_impl_wgt ( rho^{n}_{k-1/2} w^{n+1}_{k-1/2} - rho^{n}_{k+1/2} w^{n+1}_{k+1} ) / dz_{k} )
+            eta_impl = 0.5 + vwind_offctr = vwind_impl_wgt
+            eta_expl = 1.0 - eta_impl = vwind_expl_wgt
+            Note that rho^{n} is used for the implicit mass flux term.
+        exner (0:nlev-1):
+            Update exner function at full levels (cell center) from the energy equation (see eq. 3.21 or 3.25 in ICON tutorial 2023).
+            z_exner_expl = pi'^{n} - dt * rd * pi_{k} / (cvd * rho_{k} * theta_v_{k}) ( divergence(v^{n+1*} rho^{n} theta_v^{n}) + vwind_expl_wgt ( rho^{n}_{k-1/2} w^{n}_{k-1/2} theta_v^{n}_{k-1/2} - rho^{n}_{k+1/2} w^{n}_{k+1} theta_v^{n}_{k+1/2} ) / dz_{k} ) + dt * physics_tendency
+            pi^{n+1} = pi_reference + z_exner_expl - dt * vwind_impl_wgt ( rd * pi^{n} ) / ( cvd * rho^{n} * theta_v^{n} ) ( rho^{n}_{k-1/2} w^{n+1}_{k-1/2} theta_v^{n}_{k-1/2} - rho^{n}_{k+1/2} w^{n+1}_{k+1} theta_v^{n}_{k+1/2} ) / dz_{k} )
+            eta_impl = 0.5 + vwind_offctr = vwind_impl_wgt
+            eta_expl = 1.0 - eta_impl = vwind_expl_wgt
+            Note that rho^{n} and theta_v^{n} are used for the implicit flux term.
+            rho^{n}_{k-1/2} theta_v^{n}_{k-1/2} is represented by z_alpha.
+            dt * vwind_impl_wgt (rd * pi^{n}) / (cvd * rho^{n} * theta_v^{n} ) / dz_{k} is represented by z_beta.
+        theta_v (0:nlev-1):
+            Update virtual potential temperature at full levels (cell center) from the equation of state (see eqs. 3.22 and 3.23 in ICON tutorial 2023).
+            rho^{n+1} theta_v^{n+1} = rho^{n} theta_v^{n} + ( cvd * rho^{n} * theta_v^{n} ) / ( rd * pi^{n} ) ( pi^{n+1} - pi^{n} )
+        """
```

Comment:
This is a really long docstring and takes away focus from what computations are actually performed in this file. I would move it to a pure md file and place a reference here.

Same goes for other long docstrings above